zerver/views/auth.py

```diff
@@ -614,13 +617,21 @@ def sort(lst: List[UserProfile]) -> List[UserProfile]:
     context['direct_users'] = sort([u for u in users if not (u.is_realm_admin or u.is_guest)])
 
 def update_login_page_context(request: HttpRequest, context: Dict[str, Any]) -> None:
-    for key in ('email', 'already_registered', 'is_deactivated'):
+    for key in ('email', 'already_registered'):
         try:
             context[key] = request.GET[key]
         except KeyError:
             pass
 
-    context['deactivated_account_error'] = DEACTIVATED_ACCOUNT_ERROR
+    deactivated_email = request.GET.get('is_deactivated')
+    if deactivated_email is None:
+        return
+    try:
+        validate_email(deactivated_email)
+        context['deactivated_account_error'] = mark_safe(DEACTIVATED_ACCOUNT_ERROR %
+                                                         {'username': deactivated_email})
+    except ValidationError:
+        logging.info("Invalid email in is_deactivated param to login page: %s", deactivated_email)
```

Comment:
Nothing seems wrong but I'm curious why are we interested in handling this case. I understand that we show the error for something that isn't an email but I don't see any way this could happen unless someone does a GET to `?is_deactivated=<invalid email format>`(which I feel shouldn't be any concern to us) as we send `validated_email` here `return redirect_deactivated_user_to_login(realm, return_data['validated_email'])` in `social_auth_finish`.

Reply:
I'd prefer it not to be for someone to send the user to `/login/?is_deactivate=<evil message>` with `evil_message` including for example `(Please send your login and password to my_evil_email@example.com to regain access)` or something like that, leading to that being displayed inside the error message shown to the user :slightly_smiling_face:

Reply:
That's an interesting scenario to consider. Thanks!